jstreegrid.js

```diff
@@ -812,16 +829,76 @@
 				_this.redraw_node(rootNode, true);
 			});
 
-		};
+    };
+
+    this._domManipulation = null; // We'll store the column nodes in this object and an id for the grid-node that started the manipulation { id: "id of the node that started the manipulation", columns: { Key-Value-Pair col-No: Column }}
+
+    function _guid() {
+      function s4() {
+        return Math.floor((1 + Math.random()) * 0x10000)
+          .toString(16)
+          .substring(1);
+      }
+      return s4() + s4() + '-' + s4() + '-' + s4() + '-' +
+        s4() + '-' + s4() + s4() + s4();
+    }
+    /*
+     * Trys to detach the tree columns on massive dom manipulations
+     */
+    this._detachColumns = function (id) {
+      // if the columns are not detached, then detach them
+      if (this._domManipulation == null) {
+        var cols = this._gridSettings.columns || [], treecol = this._gridSettings.treecol, mw = this.midWrapper;
+        this._domManipulation = { id: id, columns: {} };
+        for (var i = 0, len = cols.length; i < len; i++) {
+          if (treecol === i) {
+            continue;
+          }
+          this._domManipulation.columns[i] = mw.children(".jstree-grid-column-" + i)[0];
+          this._domManipulation.columns[i].parentNode.removeChild(this._domManipulation.columns[i]);
+        }
+      }
+      return this._domManipulation;
+    }
+
+    this._reattachColumns = function (id) {
+      if (this._domManipulation == null) { return false; }
+      if (this._domManipulation.id === id) {
+        var cols = this._gridSettings.columns || [], treecol = this._gridSettings.treecol, mw = this.midWrapper;
+        for (var i = 0, len = cols.length; i < len; i++) {
+          if (treecol === i) {
+            continue;
+          }
+          mw[0].appendChild(this._domManipulation.columns[i]);
+        }
+        this._domManipulation = null;
+      }
+      return true;
+    }
+
+    /*
+		 * Override open_node to detach the columns before redrawing child-nodes, and do reattach them afterwarts
+		 */
+    this.open_node = function (obj, callback, animation) {
+      var id = $.isArray(obj) ? _guid() : this.get_node(obj).id;
```

Comment:
yes - because if you do a deep redraw and the root + some child-nodes are opened, redraw_node is called for the root and then for every visible child node, but you have to detach/reattach the columns just once. so the id of the first node wins.